src/server.rs

```diff
@@ -0,0 +1,475 @@
+use std::collections::hash_map::Entry;
+use std::collections::VecDeque;
+use std::fmt;
+use std::mem::take;
+use std::sync::Arc;
+use std::task::{ready, Context, Poll};
+
+use asynchronous_codec::FramedWrite;
+use blockstore::{Blockstore, BlockstoreError};
+use cid::CidGeneric;
+use fnv::{FnvHashMap, FnvHashSet};
+use futures::future::{AbortHandle, Abortable, BoxFuture};
+use futures::stream::FuturesUnordered;
+use futures::{FutureExt, SinkExt, StreamExt};
+use libp2p_core::upgrade::ReadyUpgrade;
+use libp2p_identity::PeerId;
+use libp2p_swarm::{
+    ConnectionHandlerEvent, NotifyHandler, StreamProtocol, SubstreamProtocol, ToSwarm,
+};
+use smallvec::SmallVec;
+use tracing::{debug, info, trace, warn};
+
+use crate::incoming_stream::ServerMessage;
+use crate::message::Codec;
+use crate::proto::message::{
+    mod_Message::Block as ProtoBlock, mod_Message::Wantlist as ProtoWantlist, Message,
+};
+use crate::utils::stream_protocol;
+use crate::{Event, Result, StreamRequester, ToBehaviourEvent, ToHandlerEvent};
+
+type Sink = FramedWrite<libp2p_swarm::Stream, Codec>;
+type BlockWithCid<const S: usize> = (CidGeneric<S>, Vec<u8>);
+
+#[derive(Debug)]
+pub(crate) struct ServerBehaviour<const S: usize, B>
+where
+    B: Blockstore + Send + Send,
```

Comment:
```suggestion
    B: Blockstore,
```
`Blockstore` already implies those, forgot to change beetswap after my changes to it :sweat_smile: